### PR TITLE
Faster `TableFactor::toDecisionTreeFactor`

### PR DESCRIPTION
**Pull request by varunagrawal**
I remembered that `DecisionTree` prefers the keys being ordered highest to lowest for faster speed. This PR updates the vector passed to the DecisionTree constructor such that the ordering matches the key ordering which is highest to lowest.

This update results in `toDecisionTreeFactor` going from taking 2.95 seconds to 0.81 seconds for a tree with 17 keys, which means `2^17 = 131,072` values.
It's still not great, since 17 can be hit pretty quickly, but it is a step forward.

Perhaps avoiding conversions altogether would be the way forward? Maybe subclass `DiscreteConditional` from the `TableFactor`?

### CODE REVIEW

**Comment by varunagrawal** [comment]
@dellaert the strict equality check fixed the last CI.

File: gtsam/discrete/TableFactor.cpp
Comment (dellaert) [inline gtsam/discrete/TableFactor.cpp:260]: Comment on what will happen in this block

File: gtsam/discrete/TableFactor.cpp
Comment (dellaert) [inline gtsam/discrete/TableFactor.cpp]: sort what?

File: gtsam/discrete/TableFactor.cpp
Comment (dellaert) [inline gtsam/discrete/TableFactor.cpp]: by...

**Review by dellaert — APPROVED** [review]
LGTM, just more comments in code, as not easy stuff. Also, comment on why in code. No one will come back to look at the PR comment.

**Comment by dellaert** [comment]
I’ll let you merge

**Comment by varunagrawal** [comment]
I'm working on the timing branch. I'll merge this after I have everything set up so I can see the difference between current develop and after merging this branch. Please don't merge until then.

**Pull request merged**
